### PR TITLE
disable copy construction for homestore superblock

### PR DESCRIPTION
**Pull request by zichanglai**
with copy construction, superblock will be easy to cause duplicated metablk when call write() function if it is not carefully enough. we can avoid this by disable copy construction and only allow move construction.

### CODE REVIEW

**Comment by zichanglai** [comment]
> Can you bump homstore patch version assuming it compiles successfully with homeobj?

yes, I can bump homestore patch version for this.  but homeobject need make little changes to build successfully because indextable interface is changed.

File: src/include/homestore/superblk_handler.hpp [hunk 1]
```diff
@@ -32,6 +32,30 @@ class superblk {
...
+    superblk(const superblk&) = delete;
+    superblk& operator=(const superblk&) = delete;
+
+    superblk(superblk&& rhs) noexcept
+         : m_meta_mgr_cookie(rhs.m_meta_mgr_cookie)
+           , m_raw_buf(std::move(rhs.m_raw_buf))
+           , m_sb(rhs.m_sb)
+           , m_metablk_name(std::move(rhs.m_metablk_name)) {
+	rhs.m_meta_mgr_cookie = nullptr;
+	rhs.m_sb = nullptr;
```
Comment (JacksonYao287) [inline src/include/homestore/superblk_handler.hpp:44]: why this two members is initialized in initialization list and set to nullptr here? 
they are already initialized with nullptr

File: src/include/homestore/superblk_handler.hpp [hunk 2]
```diff
@@ -32,6 +32,30 @@ class superblk {
...
+    }
+
+    superblk& operator=(superblk&& rhs) noexcept {
+        if (this != &rhs) {
+            m_meta_mgr_cookie = rhs.m_meta_mgr_cookie;
+            m_raw_buf = std::move(rhs.m_raw_buf);
+            m_sb = rhs.m_sb;
+            m_metablk_name = std::move(rhs.m_metablk_name);
+            rhs.m_meta_mgr_cookie = nullptr;
+            rhs.m_sb = nullptr;
```
Comment (JacksonYao287) [inline src/include/homestore/superblk_handler.hpp:54]: the same as above

File: src/include/homestore/superblk_handler.hpp (hunk 1 above)
Comment (zichanglai) [reply src/include/homestore/superblk_handler.hpp:44]: this is move construction

**Review by yamingk — APPROVED** [review]

**Pull request merged**